src/python/pants/core/goals/test.py

```diff
@@ -62,56 +69,61 @@ class TestResult(EngineAwareReturnType):
     stdout_digest: FileDigest
     stderr: str
     stderr_digest: FileDigest
-    address: Address
     output_setting: ShowOutput
     result_metadata: ProcessResultMetadata | None
+    description: str
 
     coverage_data: CoverageData | None = None
     # TODO: Rename this to `reports`. There is no guarantee that every language will produce
     #  XML reports, or only XML reports.
     xml_results: Snapshot | None = None
     # Any extra output (such as from plugins) that the test runner was configured to output.
     extra_output: Snapshot | None = None
+    # Path prefix to use for the extra output above when materializing it into the final dist dir.
+    extra_output_prefix: str | None = None
 
     # Prevent this class from being detected by pytest as a test class.
     __test__ = False
 
     @classmethod
-    def skip(cls, address: Address, output_setting: ShowOutput) -> TestResult:
+    def skip(cls, description: str, output_setting: ShowOutput) -> TestResult:
         return cls(
             exit_code=None,
             stdout="",
             stderr="",
             stdout_digest=EMPTY_FILE_DIGEST,
             stderr_digest=EMPTY_FILE_DIGEST,
-            address=address,
             output_setting=output_setting,
+            description=description,
+            extra_output_prefix=None,
             result_metadata=None,
         )
 
     @classmethod
     def from_fallible_process_result(
         cls,
         process_result: FallibleProcessResult,
-        address: Address,
         output_setting: ShowOutput,
         *,
+        description: str,
```

Comment:
We should maybe make this more explicit, like `partition_description`. Or even `partition_description_or_address`? I'm concerned people won't know what this refers to

Reply:
👍 I can rename it.

Really I wish it was still `Address` so I could have both `spec` and `path_safe_spec` instead of tracking 2 string fields, but it feels gross to use "fake" `Address` instances in cases where we batch tests together

Reply:
>  but it feels gross to use "fake" Address instances in cases where we batch tests together

yeah, that's the wrong approach. What we really want is an algebraic data type like a Rust enum:

```rust
enum TestResultDescription {
   Address(address),
   Batch {
      description: String,
      extra_path_prefix: String,
   },
}
```

Alas Python